internal/resource/config/localidentity/local_identity_identity_profile_data_source.go

```diff
@@ -0,0 +1,671 @@
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/id"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/resourcelink"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &localIdentityIdentityProfilesDataSource{}
+	_ datasource.DataSourceWithConfigure = &localIdentityIdentityProfilesDataSource{}
+)
+
+// Create a Administrative Account data source
+func NewLocalIdentityIdentityProfilesDataSource() datasource.DataSource {
+	return &localIdentityIdentityProfilesDataSource{}
+}
+
+// localIdentityIdentityProfilesDataSource is the datasource implementation.
+type localIdentityIdentityProfilesDataSource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type localIdentityIdentityProfilesDataSourceModel struct {
+	Id                      types.String `tfsdk:"id"`
+	CustomId                types.String `tfsdk:"custom_id"`
+	Name                    types.String `tfsdk:"name"`
+	ApcId                   types.Object `tfsdk:"apc_id"`
+	AuthSources             types.Set    `tfsdk:"auth_sources"`
+	AuthSourceUpdatePolicy  types.Object `tfsdk:"auth_source_update_policy"`
+	RegistrationEnabled     types.Bool   `tfsdk:"registration_enabled"`
+	RegistrationConfig      types.Object `tfsdk:"registration_config"`
+	ProfileConfig           types.Object `tfsdk:"profile_config"`
+	FieldConfig             types.Object `tfsdk:"field_config"`
+	EmailVerificationConfig types.Object `tfsdk:"email_verification_config"`
+	DataStoreConfig         types.Object `tfsdk:"data_store_config"`
+	ProfileEnabled          types.Bool   `tfsdk:"profile_enabled"`
+}
+
+// GetSchema defines the schema for the datasource.
+func (r *localIdentityIdentityProfilesDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	schemaDef := schema.Schema{
+		Description: "Manages Local Identity Identity Profiles",
+		Attributes: map[string]schema.Attribute{
+			"custom_id": schema.StringAttribute{
```

Comment:
We don't need custom_id in any data sources. If you need an id to call the API, just use an id attribute.